### ISSUE TITLE
Warn when an interface is marked @final

### ISSUE THREAD

**Issue by dimvar** [google/closure-compiler#442]
Currently, we allow marking an interface as final, but then allow subtypes of it (eg, extending it or implementing it). We should warn that @final on interfaces is not supported.

``` js
/**
 * @interface
 * @final
 */
function Foo() {}

/**
 * @interface
 * @extends {Foo}
 */
function Bar() {}

/**
 * @constructor
 * @implements {Foo}
 */
function Baz() {}
```
